### DISCUSSION TITLE
Join at later timecode

### DISCUSSION THREAD

**Discussion by dajix350** [Syncplay/syncplay#700] [Q&A] [answered]
Hey! I have been using Syncplay with my friends now for a few weeks and so far we have absolutely loved it, it is the best solution for watching videos together remotely by far, so thank you all so much! 

We have been running into an issue, and were wondering if you all knew any workarounds for it. When someone joins the stream after we have already stareted, they are placed at the beginning of the video. When we try to jump forwards or backwards, this does not effect them until their player caches the timecode we are jumping too. This is only a minor issue when watching short YouTube videos, but when you are watching movies that can go for multiple hours, the cache catching up can take a long time. 

Does anyone know if there is a way they can jump to a timecode before its cached and catch up with the rest of us? Maybe a way I can encode the video with ffmpeg? We all use MPV and MPV.net with Syncplay by the way.

**Comment by Et0h** [comment]
The recommended way to use Syncplay is to ensure that everyone has a local copy of the file to avoid these sorts of issues.

mpv provides better streaming sync performance than VLC so you are off to a good start, but even still it is not perfect.

https://syncplay.pl/guide/trouble/ notes: "If you are trying to play further into the video than has loaded then the media player may force you to an earlier position, causing sync problems – it may be best to wait a bit for the video to cache a bit before starting playback".

I cannot comment on how to improve stream sync performance for your use case because things are mostly handled on the mpv side rather than by Syncplay and I personally don't use streaming

From experience I know that some formats are better than others in terms of allowing a user to seek part-way through a file, but I don't know whether or not that would translate into better sync when streaming a file using Syncplay and mpv. Feel free to share the result of any experiments you do for the benefit of others.

**Comment by dajix350** [comment, marked as answer]
Found a solution, and it was embarrassingly simple. Turns out, its as easy as hosting the file on an nginx server instead of the Python http server. Just use the nginx docker container as it comes, mount the directory with your movies to the nginx html directory, and enjoy. It handles it all for you.

Suddenly you can jump around the video to any timecode, even if its not in cache yet, and people joining the room will be set at the right timecode automatically.
